### PR TITLE
feat: Un/install free sticker packs in sticker market

### LINKED ISSUE #483: support installing free sticker packs

### PR DESCRIPTION
**Pull request by emizzle**
Fixes #483.

Sticker pack details retreived from contract:
 - pack data decoded from contract response
 - data contains contentHash which, once decoded, contains an IPFS identifier
 - futher pack data in EDN format is downloaded from IPFS
 - the EDN info is decoded in to a StickerPack

List of available packs from contract are obtained separately from list of installed contracts (stored as a setting in Status-go).

Sticker market contains dynamic list of sticker packs. The sticker button shown for each pack has all states defined (in the design) for all UI states (ie bought, free, installed, pending, etc)

Add modal popup showing sticker pack details and list of stickers to be un/installed. Contains a "larger" version of the sticker pack button with many differnt UI states defined.

Uninstallation of a sticker pack removes those sticker pack's stickers from the recent sticker list and persists the list

Simplify the view model by including stickers, instead of setting an "activeStickerPackId" property. This allowed for display of sticker pack stickers to be displayed in the modal popup separately from the sticker packs shown in the market.

### CODE REVIEW

File: src/app/chat/views/sticker_pack_list.nim [hunk 1]
```diff
@@ -33,28 +38,70 @@ QtObject:
...
+      inc idx
+      let installed = if mustBeInstalled: item.installed else: true
+      if(item.pack.id == packId and installed):
+        result = idx
+        break
+
+  proc hasKey*(self: StickerPackList, packId: int): bool =
+    result = self.packs.anyIt(it.pack.id == packId)
+  
+  proc `[]`*(self: StickerPackList, packId: int): StickerPack =
```
Comment (jrainville) [inline src/app/chat/views/sticker_pack_list.nim:80]: Nice

File: ui/app/AppLayouts/Chat/components/StickerButton.qml [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    property bool hasUpdate: false
+    property bool isTimedOut: false
+    property bool hasInsufficientFunds: false
+    property bool enabled: true
+    property var icon: new Object({
+        path: "../../../img/status-logo-no-bg",
+        rotation: 0,
+        runAnimation: false
+    })
+    property string text: root.style === StickerButton.StyleType.Default ? packPrice : qsTr("Buy for " + packPrice + " SNT")
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/components/StickerButton.qml:28]: Translating with a `+` is not possible
```suggestion
    property string text: root.style === StickerButton.StyleType.Default ? packPrice : qsTr("Buy for %1 SNT").arg(packPrice )
```

File: ui/app/AppLayouts/Chat/components/StickersPopup.qml [hunk 3]
```diff
@@ -140,24 +146,34 @@ Popup {
...
-
-            RoundedIcon {
+            StickerPackIconWithIndicator {
                 id: btnHistory
-                size: 24
-                color: Style.current.darkGrey
-                imgPath: "../../../img/history_icon.svg"
+                width: 24
+                height: 24
+		selected: true
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/components/StickersPopup.qml:168]: Weird indentation here

File: ui/app/AppLayouts/Chat/components/StickersPopup.qml [hunk 4]
```diff
@@ -169,43 +185,28 @@ Popup {
...
 
-                    delegate: RoundedImage {
-                        Layout.preferredHeight: height
-                        Layout.preferredWidth: width
+                    delegate: StickerPackIconWithIndicator {
+                        id: packIconWithIndicator
+                        visible: installed
                         width: 24
                         height: 24
+                        selected: stickerPackListView.selectedPackId == packId
```
Comment (jrainville) [inline ui/app/AppLayouts/Chat/components/StickersPopup.qml:196]:
```suggestion
                        selected: stickerPackListView.selectedPackId === packId
```

File: ui/app/AppLayouts/Chat/components/StickerButton.qml (hunk 2 above)
Comment (emizzle) [reply ui/app/AppLayouts/Chat/components/StickerButton.qml]: Updated and noted, thanks!

File: ui/app/AppLayouts/Chat/components/StickersPopup.qml (hunk 3 above)
Comment (emizzle) [reply ui/app/AppLayouts/Chat/components/StickersPopup.qml]: Fixed, thanks.

File: ui/app/AppLayouts/Chat/components/StickersPopup.qml (hunk 4 above)
Comment (emizzle) [reply ui/app/AppLayouts/Chat/components/StickersPopup.qml]: Fixed, good catch!

**Comment by emizzle** [comment]
@jrainville -- all comments addressed, thank you!

**Pull request merged**